channels/offline-channel-js/src/OfflineChannel.ts

```diff
@@ -119,31 +123,40 @@ export class OfflineChannel extends BaseTelemetryPlugin implements IChannelContr
                     _taskScheduler = createTaskScheduler(createAsyncPromise, "offline channel");
                     _notificationManager = core.getNotifyMgr();
                 }
-                try {
-                    _createUrlConfig(coreConfig, core, extensions, pluginChain);
-                    let ctx = _getCoreItemCtx(coreConfig, core, extensions, pluginChain);
-                    _sender.initialize(coreConfig, core, ctx, _diagLogger, _primaryChannelId, _self._unloadHooks);
-                    if (_sender) {
-                        _senderInst = _sender.getXhrInst();
-                        _offlineListener.addListener((val)=> {
-                            if (!val.isOnline) {
-                                _sendNextBatchTimer && _sendNextBatchTimer.cancel();
-                            } else {
+                // TODO: add refresh inside timer until dependency plugin is initialized
+                scheduleTimeout(() => {
+                    try {
+                        let _dependencyPlugin = _getDependencyPlugin(coreConfig, core);
+                        // make sure that online sender is initialized
+                        if (!_isLazyInit  && _dependencyPlugin && _dependencyPlugin.isInitialized()) {
+                            _isLazyInit = true;
+                            _createUrlConfig(coreConfig, core, extensions, pluginChain);
+                            let ctx = _getCoreItemCtx(coreConfig, core, extensions, pluginChain);
+                            _sender.initialize(coreConfig, core, ctx, _diagLogger, _primaryChannelId, _self._unloadHooks);
+                            if (_sender) {
```

Comment:
nit: _sender will always be true otherwise the above line would have thrown an exception, so don't really need this if check.

Reply:
removed if (_sender)